packages/core/src/utils.ts

```diff
@@ -259,6 +263,137 @@ export async function* maskStreamTags(
   }
 }
 
+/**
+ * Resolve serialized zod output - This function takes the string output ot the `jsonSchemaToZod` function
+ * and instantiates the zod object correctly.
+ *
+ * @param obj - serialized zod object
+ * @returns resolved zod object
+ */
+export function resolveSerializedZodOutput(obj: any) {
+  return Function('z', `"use strict";return (${obj});`)(z);
+}
+
+export function isVercelTool(tool: any): tool is VercelTool {
+  return (
+    typeof tool === 'object' &&
+    tool !== null &&
+    'parameters' in tool &&
+    // Optional checks
+    (!('type' in tool) || tool.type === undefined || tool.type === 'function')
```

Comment:
What do the optional checks do? Is it that vercel tools have `parameters` and ours don't? That would make sense but I'm not following the next part

Reply:
Yeah vercel tools come with parameters, while we use inputSchema. This checks for that. The optional checks are to see if type was added, but it's optional and is usually only function or undefined so might not be necessary. I'll just remove that.